plugin/api.go

```diff
@@ -77,9 +77,21 @@ type API interface {
 	// UpdateChannel updates a channel.
 	UpdateChannel(channel *model.Channel) (*model.Channel, *model.AppError)
 
+	// AddChannelMember creates a channel membership for a user.
+	AddChannelMember(channel *model.Channel, userId string) (*model.ChannelMember, *model.AppError)
+
 	// GetChannelMember gets a channel membership for a user.
 	GetChannelMember(channelId, userId string) (*model.ChannelMember, *model.AppError)
 
+	// UpdateChannelMemberRoles updates a user's roles for a channel.
+	UpdateChannelMemberRoles(channelId, userId, newRoles string) (*model.ChannelMember, *model.AppError)
+
+	// UpdateChannelMemberNotifications updates a user's notification properties for a channel.
```

Comment:
I used the term `notifications` vs. `notify props` to be consistent with the API documentation terminology.